internal/endorser/endorser.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package endorser provides a function for generating an endorsement statement for a binary.
+package endorser
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/project-oak/transparent-release/pkg/amber"
+)
+
+// GenerateEndorsement generates an endorsement statement for the given binary hash, for the given
```

Comment:
(nit) I think in-toto and most other systems (also Docker and Ent) use the term "digest" for the noun, and "hash" only as verb. Up to you, but I think that may make things a bit more consistent.

Reply:
Thanks. I fixed as many occurrences as I could without adding too many irrelevant changes to the PR.